pkg/querier/http.go

```diff
@@ -217,6 +218,36 @@ func (q *Querier) TailHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// NewPrepopulateMiddleware creates a middleware which will parse incoming http forms.
+// This is important because some endpoints can POST x-www-form-urlencoded bodies instead of GET w/ query strings.
+func NewPrepopulateMiddleware() middleware.Interface {
+	return middleware.Func(func(next http.Handler) http.Handler {
+		return &prepop{
+			next: next,
+		}
+	})
+}
+
+type prepop struct {
```

Comment:
I think you could get rid of this and use above the http.HandlerFunc (https://golang.org/pkg/net/http/#HandlerFunc) since it's very trivial.